lib/util/gitfs/fs/index.ts

```diff
@@ -48,10 +51,15 @@ export async function writeLocalFile(
   await fs.outputFile(localFileName, fileContent);
 }
 
-export async function deleteLocalFile(fileName: string): Promise<void> {
+export async function deleteFile(fileName: string): Promise<void> {
```

Comment:
Either the name of this function was wrong, or the implementation.. Can we check? @viceice could this be by `ignoreNpmrcFile` is not working?

Reply:
maybe, would need to check this later

Reply:
`deleteLocalFile` seems inconsistent for me, since other `*LocalFile` functions act relative to `localDir`. It can be useful to have both `deleteFile` and `deleteLocalFile` functions.

Reply:
I have a concern that some of the existing calls are wrong and not deleting what they intend to. Probably because the *implementation* was wrong. We maybe don't need a deleteFile function - we shouldn't delete things that aren't in the repo localDir!

Reply:
Yes, it's pretty dangerous function. Probably we can make guard checks (and logging) for this.

Also, my next commits are complex enough, so I thought what about preliminary snapshotting for all `fs` operations in same manner as we did with HTTP and `exec()`?

Reply:
I think if we remove `deleteFile`, stop using `fs-extra` directly, and block calls to `deleteLocalFile` from starting with `../` then that should be fairly effective for not removing wrong files off the file system.

Reply:
At the another point I don't know, current changes can be good enough after real-repo testing